homeassistant/components/sonos/speaker.py

```diff
@@ -685,7 +699,17 @@ def update_group_for_uid(self, uid: str) -> None:
     @callback
     def async_update_groups(self, event: SonosEvent) -> None:
         """Handle callback for topology change event."""
-        if not hasattr(event, "zone_player_uui_ds_in_group"):
+        if xml := event.variables.get("zone_group_state"):
+            zgs = ET.fromstring(xml)
```

Comment:
It would be better if SoCo took care of parsing the XML, but this can be done later.